### PR TITLE
Cross-build for Scala.js

### PR DESCRIPTION
**Pull request by armanbilge**
Crosses the core and http4s modules for Scala.js. Closes https://github.com/LEGO/woof/issues/4. I'll add some additional comments below.

### CODE REVIEW

File: modules/core/src/main/scala/org/legogroup/woof/LogInfo.scala [hunk 1]
```diff
@@ -2,6 +2,6 @@ package org.legogroup.woof
 
 import java.io.File
 
-case class LogInfo(enclosingClass: String, file: File, lineNumber: Int):
+case class LogInfo(enclosingClass: String, fileName: String, lineNumber: Int):
```
Comment (armanbilge) [inline modules/core/src/main/scala/org/legogroup/woof/LogInfo.scala:5]: JavaScript doesn't support files (what is a file for a browser?) so neither does Scala.js.

If you _know_ your code will not be running in a browser and only on Node.js, which does have files, you can use the `fs2.io.file` package in fs2-io.
https://github.com/typelevel/fs2/blob/main/io/shared/src/main/scala/fs2/io/file/Files.scala

File: modules/core/src/main/scala/org/legogroup/woof/Logger.scala [hunk 2]
```diff
@@ -29,7 +29,9 @@ open class Logger[F[_]: StringLocal: Monad: Clock](output: Output[F], outputs: O
       message: String,
       context: List[(String, String)],
   ): F[String] =
-    Clock[F].realTimeInstant.map(now => summon[Printer].toPrint(now, level, info, message, context))
+    Clock[F].realTime
+      .map(d => Instant.ofEpochMilli(d.toMillis))
+      .map(now => summon[Printer].toPrint(now, level, info, message, context))
```
Comment (armanbilge) [inline modules/core/src/main/scala/org/legogroup/woof/Logger.scala:34]: As mentioned above, Scala.js doesn't include support for `java.time.Instant` without adding additional dependencies. So Cats Effect does not use it in its Scala.js APIs.

**Review by armanbilge — COMMENTED** [review]
I think that should clarify the changes I made. Let me know if you have any questions!

**Comment by hejfelix** [comment]
Thanks for this -- unfortunately I went through with similar changes already yesterday (see [feature/](https://github.com/LEGO/woof/tree/feature/scalajs-cross))
Compared to this, I decided to not add a dependency and instead go w. epoch millis and then some jvm and js specific code. Check out my branch, it's pretty similar!

File: build.sbt [hunk 3]
```diff
@@ -6,17 +6,20 @@ val V = new {
...
-  val munitCatsEffect = "org.typelevel" %% "munit-cats-effect-3" % V.munitCatsEffect
-  val slf4jApi        = "org.slf4j"      % "slf4j-api"           % V.slf4j
+  val cats              = Def.setting("org.typelevel" %%% "cats-core" % V.cats)
+  val catsEffect        = Def.setting("org.typelevel" %%% "cats-effect" % V.catsEffect)
+  val http4s            = Def.setting("org.http4s" %%% "http4s-core" % V.http4s)
+  val javassist         = "org.javassist" % "javassist" % V.javassist
+  val munit             = Def.setting("org.scalameta" %%% "munit" % V.munit)
+  val munitCatsEffect   = Def.setting("org.typelevel" %%% "munit-cats-effect-3" % V.munitCatsEffect)
+  val scalaJavaTime     = Def.setting("io.github.cquiroz" %%% "scala-java-time" % V.scalaJavaTime)
+  val scalaJavaTimeTzDb = Def.setting("io.github.cquiroz" %%% "scala-java-time-tzdb" % V.scalaJavaTime)
```
[the remark this answers is not in the document]
Comment (hejfelix) [reply build.sbt:21]: Ahh, nice w. `Def.setting(...)`. I'll steal this -- I wasn't able to just add the 3rd `%` like I expected :D this is much better

**Comment by hejfelix** [comment]
I'm closing this in advantage of https://github.com/LEGO/woof/pull/7 since that one doesn't add new dependencies -- I included some changes based on this. Again, thanks for doing this, your explanations were also very helpful!

**Pull request closed without merging**